lib/expand.js

```diff
@@ -43,6 +43,107 @@ class Expander {
         this.expandElement(de);
       }
     }
+
+    const entryDE = this._expanded.dataElements.find('shr.base', 'Entry');
+    let entryFields = null;
+    if (!entryDE) {
+      logger.warn('Could not find expanded definition of shr.base.Entry. Inheritance calculations will be incomplete.');
+      entryFields = {
+        'shr.core': { 'Version': true },
+        'shr.base': {}
+      };
+      for (const name of ['ShrId', 'EntryId', 'EntryType', 'FocalSubject', 'SubjectIsThirdPartyFlag', 'Narrative',
+        'Informant', 'Author', 'AssociatedEncounter', 'OriginalCreationDate', 'LastUpdateDate', 'Language']) {
+        entryFields['shr.base'][name] = true;
+      }
+    }
+    for (const de of this._expanded.dataElements.all) {
+      // TODO: What are the rules for inheriting values from multiple parents? Currently if a value or field matches any parent then it is considered inherited.
+      if (de.basedOn && de.basedOn.length) {
+        const parents = [];
+        for (let i = 0; i < de.basedOn.length; i += 1) {
```

Comment:
I usually use `i += 1` because one of my CS professors had a dislike for `i++`. However, it's moot because `for (const basedOn of de.basedOn)` is better anyways.